src/record_accessor/flb_ra_parser.c

```diff
@@ -27,6 +27,21 @@
 #include "ra_parser.h"
 #include "ra_lex.h"
 
+int flb_ra_parser_subkey_num(struct flb_ra_parser *rp)
+{
+    if (rp == NULL || rp->key == NULL) {
+        return -1;
+    }
+    else if (rp->type != FLB_RA_PARSER_KEYMAP) {
+        return 0;
+    }
+    else if(rp->key->subkeys == NULL) {
+        return -1;
+    }
+
+    return mk_list_size(rp->key->subkeys);
```

Comment:
Beware of `mk_list_size`, it iterates the entire list in order to calculate the size each time. At this point I think we might just waste 8 bytes per node and add a counter that's only used in the head node but I haven't properly assessed the impact of it yet.

Reply:
I'm glad monkey API supports that feature.

However we don't pass a head node pointer to `mk_list_del` function.
I think we need to seek the head node from a entry pointer or create a new API to pass a head node to decrement list size.

Reply:
That's true and the possible side effect could be a lot worse than the original issue, you're right.